functorch/experimental/cond.py

```diff
@@ -118,7 +106,7 @@ def cond_autograd(pred, true_fn, false_fn, *operands):
 def inner(pred, true_fn, false_fn, operands):
     mode = _get_current_dispatch_mode()
     assert (mode is not None), "Mode should always be enabled for python fallback key"
```

Comment:
nit: since this is just an assert, we shouldn't need this anymore. If there's no mode in the stack, `_pop_mode_temporarily` will fail

Reply:
I like to have it, just in case, it helped us for this refactor. It may help for future ones.